scripts/setup/nrfconnect/update_ncs.py

```diff
@@ -0,0 +1,105 @@
+
+def get_ncs_recommended_version():
+    # Read recommended version saved in the .nrfconnect-recommended-version file.
+    try:
+        with open(chip_root + 'config/nrfconnect/.nrfconnect-recommended-version', 'r') as f:
+            return f.readline().strip()
+    except:
+        raise RuntimeError("Encountered problem when trying to read .nrfconnect-recommended-version file.")
+
+def print_check_version_warning_message(current_version, recommended_version):
+    current_version_message = "WARNING: Your current NCS version (" + current_version + ")"
+    recommended_version_message = "differs from the recommended (" + recommended_version + ")."
+    allowed_message = "Please be aware that it may lead to encountering unexpected problems."
+    update_message = "Consider updating NCS to the recommended version, by calling:"
+    call_command_message = os.path.abspath(__file__) + " --update"
+
+    # Get the longest message lenght, to fit warning frame size.
+    longest_message_len = max([len(current_version_message), len(recommended_version_message), len(allowed_message), len(update_message), len(call_command_message)])
+    
+    # To keep right frame shape the space characters are added to messages shorter than the longest one.
+    print('\33[33m' + (longest_message_len+3)*'#' + '\x1b[0m')
+    print('\33[33m'+'# '+ current_version_message + (longest_message_len-len(current_version_message))*' ' + '#' + '\x1b[0m')
+    print('\33[33m'+'# '+ recommended_version_message + (longest_message_len-len(recommended_version_message))*' ' + '#' + '\x1b[0m')
+    print('\33[33m'+'# '+ longest_message_len*' '+ '#'+'\x1b[0m')
+    print('\33[33m'+'# '+ allowed_message + (longest_message_len-len(allowed_message))*' ' + '#' + '\x1b[0m')
+    print('\33[33m'+'# '+ update_message + (longest_message_len-len(update_message))*' ' + '#' + '\x1b[0m')
+    print('\33[33m'+'# '+ call_command_message + (longest_message_len-len(call_command_message))*' ' + '#' + '\x1b[0m')
+    print('\33[33m' + (longest_message_len+3)*'#' + '\x1b[0m')
+
+def main():
+
+    zephyr_base = os.getenv("ZEPHYR_BASE")
+    assert zephyr_base, "No ZEPHYR_BASE environment variable found, please set ZEPHYR_BASE to a zephyr repository path."
+
+    if args.check:
+        print("Checking current nRF Connect SDK version...")
+
+        recommended_version = get_ncs_recommended_version()
+        current_version = get_repository_commit_sha(zephyr_base + '/../nrf')
+
+        if current_version == recommended_version:
+            print("Your current version is up to date with the recommended one.")
+        else:
+            print_check_version_warning_message(current_version, recommended_version)
+
+    if args.update:
+        print("Updating nRF Connect SDK to recommended version...")
+        recommended_version = get_ncs_recommended_version()
+        update_ncs(zephyr_base + '/../nrf', recommended_version)
+
```

Comment:
I think, you should somewhere catch all the exceptions and print message they carry. The default behaviour which prints stack traces doesn't look good.